File: Algorithms/BOJ/Python/4949.py
# https://www.acmicpc.net/problem/4949
# 균형잡힌 세상
import sys
word = sys.stdin.readline().rstrip('.')
while len(word) > 2:
    flag = True
    s = []
    for c in word:
        if c in '([':
            s.append(c)
        elif c in ')':
            if s:
                if s.pop() != '(':
                    flag = False
            else:
                flag = False
        elif c in ']':
            if s:
                if s.pop() != '[':
                    flag = False
            else:
                flag = False

    if s:
        print("no")
    elif flag:
        print("yes")
    else:
        print("no")
    word = sys.stdin.readline().rstrip('.')

File: Algorithms/BOJ/Swift/4949.swift
// https://www.acmicpc.net/problem/4949
// 균형잡힌 세상
var word = readLine()!
while !word.isEmpty {
    var flag = true
    var s: [String] = []
    for c in word {
        if c == "(" || c == "[" {
            s.append(String(c))
        } else if c == ")" {
            if !s.isEmpty {
                if s.removeLast() != "(" {
                    flag = false
                }
            } else {
                flag = false
            }
        } else if c == "]" {
            if !s.isEmpty {
                if s.removeLast() != "[" {
                    flag = false
                }
            } else {
                flag = false
            }
        }
    }
    
    if !s.isEmpty {
        print("no")
    } else if flag == true {
        print("yes")
    } else if flag == false {
        print("no")
    }
    
    word = readLine()!
    word.removeLast()
}